addon/services/apollo.js

```diff
@@ -179,6 +203,23 @@ export default Service.extend({
     );
   },
 
+  managedWatchQuery(opts, resultKey, manager) {
+    let observable = this.client.watchQuery(opts);
+    manager.get('queries').pushObject(observable);
```

Comment:
There should be explicit interfaces between the service and the manager being called here, such as `addObservableQuery()` and `addSubscription()`